dataland-backend/src/main/kotlin/org/dataland/datalandbackend/api/CompanyAPI.kt

```diff
@@ -0,0 +1,107 @@
+package org.dataland.datalandbackend.api
+
+import io.swagger.v3.oas.annotations.Operation
+import io.swagger.v3.oas.annotations.responses.ApiResponse
+import io.swagger.v3.oas.annotations.responses.ApiResponses
+import org.dataland.datalandbackend.model.CompaniesRequestBody
+import org.dataland.datalandbackend.model.CompanyMetaInformation
+import org.dataland.datalandbackend.model.DataIdentifier
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RequestParam
+import javax.validation.Valid
+
+/**
+ * Defines the restful dataland-backend API regarding company data.
+ */
+
+@RequestMapping("/")
+interface CompanyAPI {
+
+    @Operation(
+        summary = "Add a new company.",
+        description = "A new company is added using the provided information, the generated company ID is returned."
+    )
+    @ApiResponses(
+        value = [
+            ApiResponse(responseCode = "200", description = "Successfully added company.")
+        ]
+    )
+    @PostMapping(
+        value = ["/companies"],
+        produces = ["application/json"],
+        consumes = ["application/json"]
+    )
+    /**
+     * A method to create a new company entry in dataland
+     * @param companyName name of the company to be created
+     * @return meta information about the stored company (id and company name)
+     */
+    fun postCompany(@Valid @RequestBody companyName: CompaniesRequestBody): ResponseEntity<CompanyMetaInformation>
+
+    @Operation(
+        summary = "Retrieve specific companies from the data store.",
+        description = "Companies identified via the provided company name are retrieved."
+    )
+    @ApiResponses(
+        value = [
+            ApiResponse(responseCode = "200", description = "Successfully retrieved a map of companies.")
+        ]
+    )
+    @GetMapping(
+        value = ["/companies"],
+        produces = ["application/json"]
+    )
+    /**
+     * A method to retrieve specific companies identified by their company names
+     * @param companyName identifier used to search for companies in the data store
+     * @return all companies whose names match with the companyName provided as search input
```

Comment:
State that all companies are returned if company name is ommited

Reply:
Fixed